enhancements/sdk-external-and-pluggable-validations.md

```diff
@@ -0,0 +1,494 @@
+        }
+    ]
+}
+```
+
+Allowing the validators to output ManifestErrors should make it easier to
+transition existing validators to the external format with minimal code.
+
+##### Migrating existing validator to executable
+
+In the short to near term, you create a new `main.go` for each of the validators.
+Then import the validator code from operator-framework/api. The `main.go` would
+take in one (1) argument, the bundle directory.
+
+Since the [existing validators](#default-validators-run-by-operator-sdk) already
+output `ManifestErrors`, it's easiest if we simply print that out as JSON to
+stdout.
+
+An example POC that takes an existing validator and outputs `ManifestErrors` can
+be found at [validator-poc][validator-poc1]
+
+Another example of a migration can be find at [ocp-olm-catalog-validator][camila-poc].
+This particular example does NOT yet output `ManifestErrors` format.
+
+TODO: do we need to add `json` tags to these structs?
+https://github.com/operator-framework/api/blob/master/pkg/validation/errors/error.go#L9-L16
+
+
+#### CLI
+
+The big question at hand is how do we indicate to the `operator-sdk` CLI that
+we want to run an external validator? Today, the `bundle validate` command takes
+in a few flags, here we will discuss how each might need to be changed to work
+with external validators.
+
+```
+Usage:
+  operator-sdk bundle validate [flags]
+
+...
+
+Flags:
+  -h, --help                                                 help for validate
+  -b, --image-builder string                                 Tool to pull and unpack bundle images. Only used when validating a bundle image. One of: [docker, podman, none] (default "docker")
+      --list-optional                                        List all optional validators available. When set, no validators will be run
+      --optional-values --optional-values=k8s-version=1.22   Inform a []string map of key=values which can be used by the validator. e.g. to check the operator bundle against an Kubernetes version that it is intended to be distributed use --optional-values=k8s-version=1.22 (default [])
+  -o, --output string                                        Result format for results. One of: [text, json-alpha1]. Note: output format types containing "alphaX" are subject to change and not covered by guarantees of stable APIs. (default "text")
+      --select-optional string                               Label selector to select optional validators to run. Run this command with '--list-optional' to list available optional validators
+
+Global Flags:
+      --plugins strings   plugin keys to be used for this subcommand execution
+      --verbose           Enable verbose logging
+```
+
+* *--help* works as is
+* *--image-builder* works as is
+* *--list-optional* would need to be updated to locate external validators.
+
+  TODO: How do we locate these validators? Do we use the XDG_CONFIG like Phase
```

Comment:
Could you please clarify the XDG_CONFIG? How does it work?

Reply:
For now, the user will pass in the path to the validators using the `--alpha-select-external` flag. We will NOT auto discover them right now.